### PR TITLE
Enable custom IMDS timeout value

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Provide an option for ManagedIdentityCredential to configure the IMDS endpoint probe timeout.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/azidentity/managed_identity_credential.go [hunk 1]
```diff
@@ -64,6 +65,10 @@ type ManagedIdentityCredentialOptions struct {
 	// instead of the hosting environment's default. The value may be the identity's client ID or resource ID, but note that
 	// some platforms don't accept resource IDs.
 	ID ManagedIDKind
+
+	// IMDSTimeout is the timeout used during initial discoverability of the IMDS endpoint.
+	// The default is one second.
+	IMDSTimeout time.Duration
```
Comment (jhendrixMSFT) [inline sdk/azidentity/managed_identity_credential.go:71]: Maybe the name could be better.

**Review by JeffreyRichter — APPROVED** [review]
You're blazingly fast man!

File: sdk/azidentity/managed_identity_credential.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azidentity/managed_identity_credential.go:72]: What about `IMDSProbeTimeout`?

**Comment by jhendrixMSFT** [comment]
Per discussion, the probe timeout only applies when using DAC.  Directly using ManagedIdentityCredential there is no timeout.

**Pull request closed without merging**